### ISSUE TITLE
version 0.90 no longer supports constructing with hashref

### ISSUE THREAD

**Issue by dakkar** [rustyconover/net-amazon-s3#86]
Moose classes can usually be instantiated both by `Class->new(foo=>'bar')` and `Class->new({ foo=>'bar' })`

The inherited `BUILDARGS` supports this, but the `BUILDARGS` added in 0.90 doesn't.

The documentation says https://metacpan.org/pod/distribution/Moose/lib/Moose/Manual/BestPractices.pod#Always-call-the-original/parent-BUILDARGS to always call the inherited `BUILDARGS`

Suggestion:
```perl
override BUILDARGS => sub {
  my $self = shift;
  my $args = super();
  # change $args as you need
  return $args
};
```

**Comment by happy-barney** [comment]
Thanks for reporting, fixed in 0.91 (with test)

**Comment by dakkar** [comment]
thank you, that was quick!
